modules/nf-core/ribowaltz/templates/ribowaltz.r

```diff
@@ -0,0 +1,479 @@
+    --gtf FILE                 Path to the GTF file. (required)
+    --fasta FILE               Path to the FASTA file. (required)
+    --length_range RANGE       Specify a range of read lengths for P-site identification, formatted as two integers separated by a colon (e.g., '26:34'). If unspecified, all lengths are considered.
+    --periodicity_threshold INT   Filter out read lengths below specified periodicity threshold for P-site identification. Must be a value between 10 and 100. (default: NULL)
+    --exclude_start INT        Number of nucleotides from start codon used to exclude P-sites near initiating ribosome when calculating CDS coverage. (default: 42)
+    --exclude_stop INT         Number of nucleotides from stop codon used to exclude P-sites near terminating ribosome when calculating CDS coverage. (default: 27)
+    -h, --help                 Display this help message and exit.
+")
+    quit(status = 0)  # Exit after displaying help
+}
+
+
+# =========
+# Parse parameters for Nextflow
+# =========
+
+# Set defaults and classes
+opt <- list(
+    output_prefix = ifelse('$task.ext.prefix' == 'null', '$meta.id', '$task.ext.prefix'),
+    bam = '$bam',
+    gtf = '$gtf',
+    fasta = '$fasta',
+    length_range = NULL,
+    periodicity_threshold = NULL,
+    exclude_start = 42,               # number of nt from start codon used to exclude P-sites near initiating ribosome when calculating CDS P-site counts.
+    exclude_stop = 27
+)
+opt_types <- lapply(opt, class)
+
+
+# Parse extra arguments
+args_opt <- parse_args('$task.ext.args')
+
+# Check if help is requested
+if ('-h' %in% args_opt || '--help' %in% args_opt) {
+    print_help_menu()
+}
+
+# Apply parameter overrides
+for ( ao in names(args_opt)){
+    if (! ao %in% names(opt)){
+        stop(paste("Invalid option:", ao))
+    } else {
+
+        # Preserve classes from defaults where possible
+        if (! is.null(opt[[ao]])){
+            args_opt[[ao]] <- as(args_opt[[ao]], opt_types[[ao]])
+        }
+        opt[[ao]] <- args_opt[[ao]]
+    }
+}
+
+# Check file inputs are valid
+for (file_input in c('bam', 'gtf', 'fasta')){
+    if (! is_valid_string(opt[[file_input]])) {
+        stop(paste("Please provide", file_input), call. = FALSE)
+    }
+
+    if (! file.exists(opt[[file_input]])){
+        stop(paste0('Value of ', file_input, ': ', opt[[file_input]], ' is not a valid file'))
+    }
+}
+
+
+# Print inputs and parsed arguments
+cat("BAM file: ", opt\$bam, "\n")
+cat("GTF file: ", opt\$gtf, "\n")
+cat("FASTA file: ", opt\$fasta, "\n")
+cat("Length range for filtering: ", opt\$length_range, "\n")
+cat("Periodicity threshold: ", opt\$periodicity_threshold, "\n")
+cat("Exclude start codons (in nt) up to: ", opt\$exclude_start, "\n")
+cat("Exclude stop codons (in nt) up to: ", opt\$exclude_stop, "\n")
+
+# Create folder for plots
+dir.create("ribowaltz_qc")
+
+
+# =========
+# Functions for riboWaltz analysis
+# =========
+
+export_offsets <- function(name, df) {
```

Comment:
Keep the functions in the same place. So move these up and apply function documentation for consistency.

Reply:
moved up and added documentation for all functions. Referencing commit 2c16f8e.